docs/working-with-projects/project-configuration/implementing-content-trust.md

```diff
@@ -3,15 +3,60 @@ title: Implementing Content Trust
 weight: 55
 ---
 
-{{< note >}}
-Notary is an optional component, please make sure you have already installed it in your Harbor instance before you go through this section.
-{{< /note >}}
+Artifact signing and signature verification are critical security capabilities that allow you to verify the integrity of an artifact. Harbor supports content trust through integrations with [Notary](https://github.com/notaryproject/notarys) and [Cosign](https://github.com/sigstore/cosign), ensuring that only signed and verified images are pulled from your Harbor instance.
+
+This page outlines Harbor's integration with Cosign and Notary as well as how to [enforce content trust](#enforce-content-trust) using default Harbor deployment policy.
+
+## Use Cosign to sign artifacts
+
+Harbor v2.5 integrates support for [Cosign](https://github.com/sigstore/cosign), a OCI artifact signing and verification solution that is part of the [Sigstore project](https://github.com/sigstore).
+
+Cosign signs OCI artifacts and pushes the generated signature into Harbor. This signature is stored as an artifact accessory along side the signed artifact. Harbor treats Cosign signatures like any other Harbor artifact, meaning you can use all of Harbor's artifact functionality to manage cosign signatures within your projects. Harbor manages the link between the signed artifact and cosign signature, allowing you to apply things like retention and immutable policies to a sign artifact, and it will extend to both. Additionally Harbor's [replication capabilities](../../administration/configuring-replication/) will replicate signatures along with signed artifacts when replicating between Harbor replication endpoints.
+
+Before starting to sign with Cosign, you must have cosign installed locally and have a generated a private key. See the [Cosign documentation](https://github.com/sigstore/cosign) for more installation information.
+
+
+### Sign, upload, and view Cosign signatures in Harbor
+
+Use the `cosign sign` command to sign an image and upload the Cosign signature to your Harbor instance. Replace `<harbor-instance>/<image/path><image-tag>` in the example below with your Harbor instance, the path to the image, and image tag.
+
+```
+cosign sign --key cosign.key <harbor-instance>/<image/path>:<image-tag>
+```
+
+After entering the password for your cosign private key, cosign will sign the image and upload the generated signature to your Harbor instance. You can view all signatures for an sign artifact in the Harbor interface.
+
+1. Log into the Harbor interface and navigate to the Project where your signed artifact is located.
+
+    [SCREENSHOT OF PROJECT PAGE]
+
+1. Expand the artifact to view the signatures attached to the image.
+
+    [SCREENSHOT OF ARTIFACT EXPANDED & ACCESSORY TABLE]
+
+### Delete Cosign signatures
```

Comment:
we can also add comments that Harbor doesn't support `cosign clean` to remove signature since it has to implement tag delete API which is not required by OCI distribution spec.

Reply:
is the tag delete API a part of cosign, or sigstore? just want to make sure i understand how its connected :)

Reply:
This is what I mean, https://github.com/opencontainers/distribution-spec/blob/main/spec.md#content-management

Harbor doesn't choose to imple tag deletion, but `cosign clean` is using it.

Reply:
I've updated the docs to include this information